src/scrollspy.ts

```diff
@@ -37,32 +37,34 @@ export interface ScrollSpyOptions extends BaseOptions {
    * @default null (browser's native animation implementation/duration)
    */
   animationDuration: number | null;
-};
+}
 
 const _defaults: ScrollSpyOptions = {
   throttle: 100,
   scrollOffset: 200, // offset - 200 allows elements near bottom of page to scroll
   activeClass: 'active',
-  getActiveElement: (id: string): string => { return 'a[href="#' + id + '"]'; },
+  getActiveElement: (id: string): string => {
+    return 'a[href="#' + id + '"]';
+  },
   keepTopElementActive: false,
-  animationDuration: null,
+  animationDuration: null
 };
 
 export class ScrollSpy extends Component<ScrollSpyOptions> {
   static _elements: ScrollSpy[];
   static _count: number;
   static _increment: number;
   static _elementsInView: ScrollSpy[];
-  static _visibleElements: any[];
+  static _visibleElements: HTMLElement[];
   static _ticks: number;
   static _keptTopActiveElement: HTMLElement | null = null;
 
   private tickId: number;
-  private id: any;
+  private id: string;
 
   constructor(el: HTMLElement, options: Partial<ScrollSpyOptions>) {
     super(el, options, ScrollSpy);
-    (this.el as any).M_ScrollSpy = this;
+    this.el['M_ScrollSpy'] = this;
```

Comment:
I think we might need to revert this, since it introduces backwards compatibility issues with applications using the previous object orientated approach, propose to just set it to warnings until we can declare the correct type definition